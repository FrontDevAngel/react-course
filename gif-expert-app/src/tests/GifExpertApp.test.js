import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas a componente <GifExpertApp />', () => {
    test('Hacer match del componente con el SnapShot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['√Åmerica', 'Chivas'];

        const wrapper = shallow(
            <GifExpertApp defaultCategories={categories} />,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
