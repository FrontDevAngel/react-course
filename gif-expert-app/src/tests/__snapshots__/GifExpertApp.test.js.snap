// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Ámerica"
      key="Ámerica"
    />
    <GifGrid
      category="Chivas"
      key="Chivas"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas a componente <GifExpertApp /> Hacer match del componente con el SnapShot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
